src/sdp.ts

```diff
@@ -1,109 +1,119 @@
-import {inappropriateMultiaddr, invalidArgument, unsupportedHashAlgorithm} from './error.js';
-import {logger} from '@libp2p/logger';
-import {Multiaddr} from '@multiformats/multiaddr';
-import * as multihashes from 'multihashes';
-import {bases} from 'multiformats/basics';
+import { inappropriateMultiaddr, invalidArgument, invalidFingerprint, unsupportedHashAlgorithm } from './error'
+import { logger } from '@libp2p/logger'
+import { Multiaddr } from '@multiformats/multiaddr'
+import * as multihashes from 'multihashes'
+import { bases } from 'multiformats/basics'
 
-const log = logger('libp2p:webrtc:sdp');
+const log = logger('libp2p:webrtc:sdp')
+const CERTHASH_CODE: number = 466
 
+// Get base2 | identity decoders
 export const mbdecoder: any = (function () {
-  const decoders = Object.values(bases).map((b) => b.decoder);
-  let acc = decoders[0].or(decoders[1]);
-  decoders.slice(2).forEach((d) => (acc = acc.or(d)));
-  return acc;
-})();
+  const decoders = Object.values(bases).map((b) => b.decoder)
+  let acc = decoders[0].or(decoders[1])
+  decoders.slice(2).forEach((d) => (acc = acc.or(d)))
+  return acc
+})()
 
-const CERTHASH_CODE: number = 466;
-
-function ipv(ma: Multiaddr): string {
+// Extract the ipv from a multiaddr
+function ipv (ma: Multiaddr): string {
   for (const proto of ma.protoNames()) {
     if (proto.startsWith('ip')) {
-      return proto.toUpperCase();
+      return proto.toUpperCase()
     }
   }
-  log('Warning: multiaddr does not appear to contain IP4 or IP6.', ma);
-  return 'IP6';
-}
-function ip(ma: Multiaddr): string {
-  return ma.toOptions().host;
-}
-function port(ma: Multiaddr): number {
-  return ma.toOptions().port;
+
+  log('Warning: multiaddr does not appear to contain IP4 or IP6.', ma)
+
+  return 'IP6'
 }
 
-export function certhash(ma: Multiaddr): string {
-  const tups = ma.stringTuples();
-  const certhash_value = tups.filter((tup) => tup[0] == CERTHASH_CODE).map((tup) => tup[1])[0];
-  if (certhash_value) {
-    return certhash_value;
-  } else {
-    throw inappropriateMultiaddr("Couldn't find a certhash component of multiaddr:" + ma.toString());
+// Extract the certhash from a multiaddr
+export function certhash (ma: Multiaddr): string {
+  const tups = ma.stringTuples()
+  const certhash = tups.filter((tup) => tup[0] === CERTHASH_CODE).map((tup) => tup[1])[0]
+
+  if (certhash === undefined || certhash === '') {
+    throw inappropriateMultiaddr(`Couldn't find a certhash component of multiaddr: ${ma.toString()}`)
   }
+
+  return certhash
 }
 
-export function decodeCerthash(certhash: string) {
-  const mbdecoded = mbdecoder.decode(certhash);
-  return multihashes.decode(mbdecoded);
+// Convert a certhash into a multihash
+export function decodeCerthash (certhash: string) {
+  const mbdecoded = mbdecoder.decode(certhash)
+  return multihashes.decode(mbdecoded)
 }
 
-export function certhashToFingerprint(ma: Multiaddr): string[] {
+// Extract the fingerprint from a multiaddr
```

Comment:
comments are imho only useful if they serve a purpose. These simply state what the naming should state. What is less clear (sometimes) and can (if you want) give value is define how it extracts this info. E.g. why the decoding, why the reducing and transformations that happen, etc etc.... With some examples and what not.

Reply:
The goal is to fully document all public functions.  I had to stop myself from a full refactor b/c this should mainly be  a linting PR.  I'll add a note for this in the PR description.

Reply:
Description updated.

Reply:
In Go they also force you to document all public things, but IMHO it usually leads to a strain on the developer and a bunch of useless comments. Clean code structure and proper naming is usually enough to cover this. While the comments are better served as last resort (IMHO) and for examples + documenting things code cannot explain.

But I guess this is partly out of our control and partly out of scope for this PR. So nvm.